spec/models/vm_or_template_spec.rb

```diff
@@ -471,4 +471,34 @@
       expect(VmOrTemplate.batch_operation_supported?(:migrate, [vm1.id, vm2.id])).to eq(true)
     end
   end
+
+  context ".set_tenant_from_group" do
+    before { EvmSpecHelper.create_root_tenant }
+    let(:tenant) { FactoryGirl.create(:tenant, :parent => Tenant.root_tenant) }
+    let(:tenant2) { FactoryGirl.create(:tenant, :parent => Tenant.root_tenant) }
+    let(:group) { FactoryGirl.create(:miq_group, :tenant => tenant) }
+    let(:group2) { FactoryGirl.create(:miq_group, :tenant => tenant2) }
+
+    it "assigns the tenant from the group" do
+      expect(FactoryGirl.create(:vm_vmware, :miq_group => group).tenant).to eq(tenant)
+    end
+
+    it "assigns the tenant from the group_id" do
+      expect(FactoryGirl.create(:vm_vmware, :miq_group_id => group.id).tenant).to eq(tenant)
+    end
+
+    it "assigns the tenant from the group over the tenant" do
+      expect(FactoryGirl.create(:vm_vmware, :miq_group => group, :tenant => tenant2).tenant).to eq(tenant)
```

Comment:
Should this raise an exception or fail a validation?  Would this be the expected behavior?

By the way, can you change your lets so tenant methods invoked here are clear?  (one is from the let, one is the message sent to the vm)  Maybe change the let to `:tenant1`